Test/Microsoft.ApplicationInsights.Test/Shared/DataContracts/DependencyTelemetryTest.cs

```diff
@@ -74,6 +80,35 @@ public void RemoteDependencyTelemetrySerializesToJson()
             AssertEx.AreEqual(expected.Properties.ToArray(), item.data.baseData.properties.ToArray());
         }
 
+        public void RemoteDependencySerializationPicksUpCorrectState()
+        {
+            DependencyTelemetry expected = this.CreateRemoteDependencyTelemetry();
+            ((ITelemetry)expected).Sanitize();
+            byte[] buf = new byte[1000000];
+            expected.SerializeData(new JsonSerializationWriter(new StreamWriter(new MemoryStream(buf))));
+
+            // Change the telemetry after serialization.
+            expected.Name = expected.Name + "new";
+
+            // Validate that the newly updated Name is picked up.
+            var item = TelemetryItemTestHelper.SerializeDeserializeTelemetryItem<AI.RemoteDependencyData>(expected);
+
+            Assert.AreEqual<DateTimeOffset>(expected.Timestamp, DateTimeOffset.Parse(item.time, null, System.Globalization.DateTimeStyles.AssumeUniversal));
+            Assert.AreEqual(expected.Sequence, item.seq);
+            Assert.AreEqual(expected.Context.InstrumentationKey, item.iKey);
+            AssertEx.AreEqual(expected.Context.SanitizedTags.ToArray(), item.tags.ToArray());
+            Assert.AreEqual(nameof(AI.RemoteDependencyData), item.data.baseType);
+
+            Assert.AreEqual(expected.Id, item.data.baseData.id);
+            Assert.AreEqual(expected.ResultCode, item.data.baseData.resultCode);
+            Assert.AreEqual(expected.Name, item.data.baseData.name);
+            Assert.AreEqual(expected.Duration, TimeSpan.Parse(item.data.baseData.duration));
+            Assert.AreEqual(expected.Type, item.data.baseData.type);
+
+            Assert.AreEqual(expected.Success, item.data.baseData.success);
+            AssertEx.AreEqual(expected.Properties.ToArray(), item.data.baseData.properties.ToArray());
```

Comment:
This will need a change / merge as soon as #1001 is merged. Most likely, expected.Properties will contain Extension as a flat list.